packages/components/src/form-token-field/test/index.tsx

```diff
@@ -0,0 +1,2089 @@
+			// result as before — no tokens are deleted.
+			await user.type( input, '[ArrowRight][Delete]' );
+
+			// Proof that so far, all keyboard interactions didn't delete any tokens.
+			expect( onChangeSpy ).not.toHaveBeenCalled();
+			expectTokensToBeInTheDocument( [
+				'kiwi',
+				'peach',
+				'nectarine',
+				'coconut',
+			] );
+
+			// Press the left arrow 4 times, moving cursor between the "kiwi" and
+			// "peach" tokens. Pressing the "delete" key will delete the "peach"
+			// token, since it's in front of the cursor.
+			await user.type(
+				input,
+				'[ArrowLeft][ArrowLeft][ArrowLeft][ArrowLeft][Delete]'
+			);
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+			expect( onChangeSpy ).toHaveBeenCalledWith( [
+				'peach',
+				'nectarine',
+				'coconut',
+			] );
+			expectTokensToBeInTheDocument( [
+				'peach',
+				'nectarine',
+				'coconut',
+			] );
+			expectTokensNotToBeInTheDocument( [ 'kiwi' ] );
+
+			// Press backspace to delete the token before the cursor, but since
+			// there's no token before the cursor, nothing happens
+			await user.type( input, '[Backspace]' );
+
+			// Pressing the left arrow doesn't move the cursor because there are no
+			// tokens before it, and therefore pressing backspace yields the same
+			// result as before — no tokens are deleted.
+			await user.type( input, '[ArrowLeft][Backspace]' );
+
+			// Proof that pressing backspace hasn't caused any further token deletion.
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+
+			// Press the right arrow, moving cursor between the "kiwi" and
+			// "nectarine" tokens. Pressing the "delete" key will delete the "nectarine"
+			// token, since it's in front of the cursor.
+			await user.type( input, '[ArrowRight][Delete]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 2 );
+			expect( onChangeSpy ).toHaveBeenCalledWith( [
+				'peach',
+				'coconut',
+			] );
+			expectTokensToBeInTheDocument( [ 'peach', 'coconut' ] );
+			expectTokensNotToBeInTheDocument( [ 'kiwi', 'nectarine' ] );
+
+			// Add 'starfruit' token while the cursor is in between the "peach" and
+			// "coconut" tokens.
+			await user.type( input, 'starfruit[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 3 );
+			expect( onChangeSpy ).toHaveBeenCalledWith( [
+				'peach',
+				// Notice that starfruit is added in between "peach" and "coconut"
+				'starfruit',
+				'coconut',
+			] );
+			expectTokensToBeInTheDocument( [
+				'peach',
+				'starfruit',
+				'coconut',
+			] );
+		} );
+
+		it( "should add additional classnames passed via the `className` prop to the input element's 2nd level wrapper", () => {
+			render( <FormTokenFieldWithState className="test-classname" /> );
+
+			const input = screen.getByRole( 'combobox' );
+
+			// This is testing implementation details, but I'm not sure there's
+			// a better way.
+			expect( input.parentElement?.parentElement ).toHaveClass(
```

Comment:
Alternatively, we could go from the top to the bottom, but I think it's the same and is still better than querying by classname.

Reply:
Yeah agreed, we can probably leave it as-is